backend/fms_core/models/derived_sample.py

```diff
@@ -120,6 +120,9 @@ class DerivedSample(TrackedModel):
     library = models.CharField(max_length=200, blank=True, null=True)
     index = models.CharField(max_length=200, blank=True, null=True)
 
+    quality_flag = models.CharField(choices=[('Passed', 'Passed'), ('Failed', 'Failed')], help_text='Quality flag of the sample.', max_length=20)
```

Comment:
This would be better represented by a boolean i think (True, "Passed"), (False, "Failed"). Make sure it's Nullable to be able to tell the difference from an unset flag and a false flag.